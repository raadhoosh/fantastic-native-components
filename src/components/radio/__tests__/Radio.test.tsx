import Radio from "../index";
import { shallow, ShallowWrapper } from "enzyme";
import React from "react";

const createTestProps = (props: Object) => ({
  ...props,
});

describe("Radio", () => {
  describe("rendering", () => {
    let component: ShallowWrapper;    
    beforeEach(() => {      
      component = shallow(<Radio primary 
        color= '#f00'
        fontSize={12} 
        textColor="#9c27b0"
        label="secondary"
        checked={true}
        onPress={() => {
          alert('You tapped the Radio!');
      }}
      />);
    });
    it("renders", () => {
      expect(component.exists()).toBe(true);
  });
  it("toMatchSnapshot", () => {
      expect(component).toMatchSnapshot();
  });
  });
});
