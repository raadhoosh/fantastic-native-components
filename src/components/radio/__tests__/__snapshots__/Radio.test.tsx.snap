// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Radio rendering toMatchSnapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Radio
    checked={true}
    color="#f00"
    fontSize={12}
    label="secondary"
    onPress={[Function]}
    primary={true}
    textColor="#9c27b0"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        <ForwardRef
          checked={true}
          color="#f00"
          fontSize={12}
          label="secondary"
          onPress={[Function]}
          primary={true}
          textColor="#9c27b0"
        >
          <ForwardRef
            checked={true}
            color="#f00"
            fontSize={12}
            label="secondary"
            name="circle"
            onPress={[Function]}
            primary={true}
            size={11}
            textColor="#9c27b0"
            type="FontAwesome"
          />
        </ForwardRef>,
        <ForwardRef
          fontSize={12}
          textColor="#9c27b0"
        >
          secondary
        </ForwardRef>,
      ],
      "onPress": [Function],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "checked": true,
          "children": <ForwardRef
            checked={true}
            color="#f00"
            fontSize={12}
            label="secondary"
            name="circle"
            onPress={[Function]}
            primary={true}
            size={11}
            textColor="#9c27b0"
            type="FontAwesome"
          />,
          "color": "#f00",
          "fontSize": 12,
          "label": "secondary",
          "onPress": [Function],
          "primary": true,
          "textColor": "#9c27b0",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "checked": true,
            "color": "#f00",
            "fontSize": 12,
            "label": "secondary",
            "name": "circle",
            "onPress": [Function],
            "primary": true,
            "size": 11,
            "textColor": "#9c27b0",
            "type": "FontAwesome",
          },
          "ref": null,
          "rendered": null,
          "type": Object {
            "$$typeof": Symbol(react.forward_ref),
            "displayName": "WithTheme(Icon)",
            "render": [Function],
            Symbol(Symbol.hasInstance): [Function],
          },
        },
        "type": Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "sc-jDwBTQ",
            "isStatic": false,
            "rules": Array [
              "      
    border: 2px solid ",
              [Function],
              ";        
    width: ",
              [Function],
              ";       
    height: ",
              [Function],
              ";             
    border-radius: 500;     
    justify-content: center;
    align-items: center;
    background-color: #fff;
    margin-bottom: 5; 
    margin-right: 5;   
",
            ],
          },
          "displayName": "Styled(Component)",
          "render": [Function],
          "styledComponentId": "sc-jDwBTQ",
          "target": [Function],
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
          Symbol(Symbol.hasInstance): [Function],
        },
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": "secondary",
          "fontSize": 12,
          "textColor": "#9c27b0",
        },
        "ref": null,
        "rendered": "secondary",
        "type": Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "sc-gPEVay",
            "isStatic": false,
            "rules": Array [
              "
   color: ",
              [Function],
              ";    
   font-size: ",
              [Function],
              ";    
   flex: 1;
   align-items: center;
   justify-content: center;
   height: 100%;
",
            ],
          },
          "displayName": "Styled(Component)",
          "propTypes": Object {
            "accessible": [Function],
            "adjustsFontSizeToFit": [Function],
            "allowFontScaling": [Function],
            "disabled": [Function],
            "ellipsizeMode": [Function],
            "minimumFontScale": [Function],
            "nativeID": [Function],
            "numberOfLines": [Function],
            "onLayout": [Function],
            "onLongPress": [Function],
            "onPress": [Function],
            "pressRetentionOffset": [Function],
            "selectable": [Function],
            "selectionColor": [Function],
            "style": [Function],
            "suppressHighlighting": [Function],
            "testID": [Function],
            "textBreakStrategy": [Function],
          },
          "render": [Function],
          "styledComponentId": "sc-gPEVay",
          "target": [Function],
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
          Symbol(Symbol.hasInstance): [Function],
        },
      },
    ],
    "type": Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": Array [],
      "componentStyle": ComponentStyle {
        "componentId": "sc-jAaTju",
        "isStatic": true,
        "rules": Array [
          "      
   flex: 1;
   flex-direction: row;
   align-items: center;   
",
        ],
      },
      "displayName": "Styled(TouchableOpacity)",
      "getDefaultProps": [Function],
      "render": [Function],
      "styledComponentId": "sc-jAaTju",
      "target": [Function],
      "toString": [Function],
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
      Symbol(Symbol.hasInstance): [Function],
    },
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <ForwardRef
            checked={true}
            color="#f00"
            fontSize={12}
            label="secondary"
            onPress={[Function]}
            primary={true}
            textColor="#9c27b0"
          >
            <ForwardRef
              checked={true}
              color="#f00"
              fontSize={12}
              label="secondary"
              name="circle"
              onPress={[Function]}
              primary={true}
              size={11}
              textColor="#9c27b0"
              type="FontAwesome"
            />
          </ForwardRef>,
          <ForwardRef
            fontSize={12}
            textColor="#9c27b0"
          >
            secondary
          </ForwardRef>,
        ],
        "onPress": [Function],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "checked": true,
            "children": <ForwardRef
              checked={true}
              color="#f00"
              fontSize={12}
              label="secondary"
              name="circle"
              onPress={[Function]}
              primary={true}
              size={11}
              textColor="#9c27b0"
              type="FontAwesome"
            />,
            "color": "#f00",
            "fontSize": 12,
            "label": "secondary",
            "onPress": [Function],
            "primary": true,
            "textColor": "#9c27b0",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "checked": true,
              "color": "#f00",
              "fontSize": 12,
              "label": "secondary",
              "name": "circle",
              "onPress": [Function],
              "primary": true,
              "size": 11,
              "textColor": "#9c27b0",
              "type": "FontAwesome",
            },
            "ref": null,
            "rendered": null,
            "type": Object {
              "$$typeof": Symbol(react.forward_ref),
              "displayName": "WithTheme(Icon)",
              "render": [Function],
              Symbol(Symbol.hasInstance): [Function],
            },
          },
          "type": Object {
            "$$typeof": Symbol(react.forward_ref),
            "attrs": Array [],
            "componentStyle": ComponentStyle {
              "componentId": "sc-jDwBTQ",
              "isStatic": false,
              "rules": Array [
                "      
    border: 2px solid ",
                [Function],
                ";        
    width: ",
                [Function],
                ";       
    height: ",
                [Function],
                ";             
    border-radius: 500;     
    justify-content: center;
    align-items: center;
    background-color: #fff;
    margin-bottom: 5; 
    margin-right: 5;   
",
              ],
            },
            "displayName": "Styled(Component)",
            "render": [Function],
            "styledComponentId": "sc-jDwBTQ",
            "target": [Function],
            "toString": [Function],
            "warnTooManyClasses": [Function],
            "withComponent": [Function],
            Symbol(Symbol.hasInstance): [Function],
          },
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": "secondary",
            "fontSize": 12,
            "textColor": "#9c27b0",
          },
          "ref": null,
          "rendered": "secondary",
          "type": Object {
            "$$typeof": Symbol(react.forward_ref),
            "attrs": Array [],
            "componentStyle": ComponentStyle {
              "componentId": "sc-gPEVay",
              "isStatic": false,
              "rules": Array [
                "
   color: ",
                [Function],
                ";    
   font-size: ",
                [Function],
                ";    
   flex: 1;
   align-items: center;
   justify-content: center;
   height: 100%;
",
              ],
            },
            "displayName": "Styled(Component)",
            "propTypes": Object {
              "accessible": [Function],
              "adjustsFontSizeToFit": [Function],
              "allowFontScaling": [Function],
              "disabled": [Function],
              "ellipsizeMode": [Function],
              "minimumFontScale": [Function],
              "nativeID": [Function],
              "numberOfLines": [Function],
              "onLayout": [Function],
              "onLongPress": [Function],
              "onPress": [Function],
              "pressRetentionOffset": [Function],
              "selectable": [Function],
              "selectionColor": [Function],
              "style": [Function],
              "suppressHighlighting": [Function],
              "testID": [Function],
              "textBreakStrategy": [Function],
            },
            "render": [Function],
            "styledComponentId": "sc-gPEVay",
            "target": [Function],
            "toString": [Function],
            "warnTooManyClasses": [Function],
            "withComponent": [Function],
            Symbol(Symbol.hasInstance): [Function],
          },
        },
      ],
      "type": Object {
        "$$typeof": Symbol(react.forward_ref),
        "attrs": Array [],
        "componentStyle": ComponentStyle {
          "componentId": "sc-jAaTju",
          "isStatic": true,
          "rules": Array [
            "      
   flex: 1;
   flex-direction: row;
   align-items: center;   
",
          ],
        },
        "displayName": "Styled(TouchableOpacity)",
        "getDefaultProps": [Function],
        "render": [Function],
        "styledComponentId": "sc-jAaTju",
        "target": [Function],
        "toString": [Function],
        "warnTooManyClasses": [Function],
        "withComponent": [Function],
        Symbol(Symbol.hasInstance): [Function],
      },
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
